crates/ruff_linter/src/rules/flake8_return/rules/function.rs

```diff
@@ -440,20 +440,45 @@ fn is_noreturn_func(func: &Expr, semantic: &SemanticModel) -> bool {
     semantic.match_typing_qualified_name(&qualified_name, "NoReturn")
 }
 
-/// RET503
-fn implicit_return(checker: &mut Checker, stmt: &Stmt) {
+fn add_return_none(checker: &mut Checker, stmt: &Stmt, range: TextRange) {
+    let mut diagnostic = Diagnostic::new(ImplicitReturn, range);
+    if let Some(indent) = indentation(checker.locator(), stmt) {
+        let mut content = String::new();
+        content.push_str(checker.stylist().line_ending().as_str());
+        content.push_str(indent);
+        content.push_str("return None");
+        diagnostic.set_fix(Fix::unsafe_edit(Edit::insertion(
+            content,
+            end_of_last_statement(stmt, checker.locator()),
+        )));
+    }
+    checker.diagnostics.push(diagnostic);
+}
+
+fn has_implicit_return(checker: &mut Checker, stmt: &Stmt) -> bool {
     match stmt {
         Stmt::If(ast::StmtIf {
             body,
             elif_else_clauses,
             ..
         }) => {
+            let mut result = false;
             if let Some(last_stmt) = body.last() {
-                implicit_return(checker, last_stmt);
+                if has_implicit_return(checker, last_stmt) {
+                    result = true;
+                    if checker.settings.preview.is_enabled() {
+                        return result;
+                    }
```

Comment:
Early return if in preview mode because we only need to know if the function has at least one implicit return to add a single `return None` at the end of the function.
The stable version can add multiple return so it cannot return early.
The code should be simplified once the preview version become stable.